plugins/processors/ec2tagger/ec2tagger.go

```diff
@@ -309,25 +242,22 @@ func (t *Tagger) Init() error {
 			t.metadataLookup.instanceId = true
 		case mdKeyImageId:
 			t.metadataLookup.imageId = true
-		case mdKeyInstaneType:
+		case mdKeyInstanceType:
 			t.metadataLookup.instanceType = true
 		default:
-			t.Log.Errorf("ec2tagger: Unsupported EC2 Metadata key: %s", tag)
+			t.Log.Errorf("ec2tagger: Unsupported EC2 Metadata key: %s.", tag)
 		}
 	}
 
-	t.Log.Infof("ec2tagger: Check ec2 metadata")
-	if !t.ec2metadata.Available() {
-		msg := "ec2tagger: Unable to retrieve InstanceId. This plugin must only be used on an EC2 instance"
-		t.Log.Errorf(msg)
-		return errors.New(msg)
-	}
+	t.Log.Infof("ec2tagger: Check EC2 Metadata.")
 
-	doc, err := t.ec2metadata.GetInstanceIdentityDocument()
-	if nil != err {
-		msg := fmt.Sprintf("ec2tagger: Unable to retrieve InstanceId : %+v", err.Error())
-		t.Log.Errorf(msg)
-		return errors.New(msg)
+	doc, err := t.ec2MetadataProvider().GetInstanceIdentityDocument()
+	if err != nil {
+		t.Log.Error("ec2tagger: Timeout when retrieving EC2 Metadata. This plugin must only be used on an EC2 instance.")
+		if context.CurrentContext().RunInContainer() {
+			t.Log.Warn("ec2tagger: Please increase hop limit to 2 by following this document https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-options.html#configuring-IMDS-existing-instances.")
```

Comment:
ec2tagger: timeout may occurred because hop limit is too small. Please increase hop limit to 2 by following this document https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-options.html#configuring-IMDS-existing-instances.

Reply:
Agree 👍 . Thanks for the suggesting wording.